indico/modules/events/cloning_test.py

```diff
@@ -0,0 +1,92 @@
+     [datetime(2017, 2, 28, 3, 0, 0), datetime(2017, 3, 1, 3, 0, 0), datetime(2017, 3, 2, 3, 0, 0)], False)
+))
+def test_interval_clone(start_dt, delta, stop_criterion, num_times, until_dt,
+                        expected_dates, expected_flag, dummy_event):
+    dummy_event.timezone = 'Europe/Zurich'
+    clone_calculator = IntervalCloneCalculator(dummy_event)
+    form = CloneRepeatIntervalForm(dummy_event, csrf_enabled=False, recurrence=delta, stop_criterion=stop_criterion,
+                                   num_times=num_times)
+    expected_dates = [dummy_event.tzinfo.localize(dt) for dt in expected_dates]
+    form.start_dt.data = dummy_event.tzinfo.localize(start_dt)
+    form.until_dt.data = dummy_event.tzinfo.localize(until_dt) if until_dt else None
+    dates, last_day_of_the_month = clone_calculator._calculate(form)
+    assert last_day_of_the_month is expected_flag
+    assert dates == expected_dates
+
+
+@pytest.mark.parametrize(('start_dt', 'week_day', 'num_months',  'stop_criterion', 'num_times', 'until_dt',
+                          'expected_dates', 'expected_flag'), (
+    (datetime(2017, 12, 7, 3, 0, 0), (1, 0), 1, 'num_times', 2, None,
+     [datetime(2018, 1, 1, 3, 0, 0), datetime(2018, 2, 5, 3, 0, 0)], False),
+    (datetime(2018, 1, 1, 3, 0, 0), (1, 0), 1, 'num_times', 2, None,
+     [datetime(2018, 1, 1, 3, 0, 0), datetime(2018, 2, 5, 3, 0, 0)], False),
+    (datetime(2017, 11, 25, 3, 0, 0), (4, 6), 1, 'num_times', 3, None,
+     [datetime(2017, 11, 26, 3, 0, 0), datetime(2017, 12, 24, 3, 0, 0), datetime(2018, 1, 28, 3, 0, 0)], False),
+    (datetime(2017, 11, 25, 3, 0, 0), (-1, 6), 1, 'num_times', 3, None,
+     [datetime(2017, 11, 26, 3, 0, 0), datetime(2017, 12, 31, 3, 0, 0), datetime(2018, 1, 28, 3, 0, 0)], False),
+    (datetime(2017, 7, 19, 3, 0, 0), (3, 2), 2, 'day', None, datetime(2017, 11, 30, 3, 0, 0),
+     [datetime(2017, 7, 19, 3, 0, 0), datetime(2017, 9, 20, 3, 0, 0), datetime(2017, 11, 15, 3, 0, 0)], False),
+
+))
+def test_pattern_clone(start_dt, week_day, num_months, stop_criterion, num_times, until_dt, expected_dates,
+                       expected_flag, dummy_event):
+    dummy_event.timezone = 'Europe/Zurich'
+    clone_calculator = PatternCloneCalculator(dummy_event)
+    form = CloneRepeatPatternForm(dummy_event, csrf_enabled=False, week_day=week_day, num_months=num_months,
+                                  stop_criterion=stop_criterion, num_times=num_times, until_dt=until_dt)
+    expected_dates = [dummy_event.tzinfo.localize(dt) for dt in expected_dates]
+    form.start_dt.data = dummy_event.tzinfo.localize(start_dt)
+    dates, last_day_of_the_month = clone_calculator._calculate(form)
+    assert last_day_of_the_month is expected_flag
+    assert dates == expected_dates
+
+
+@pytest.mark.parametrize(('event_date', 'event_timezone'), (
+    (datetime(2018, 03, 25, 2, 0, 0), 'Europe/Zurich'),
```

Comment:
leading zeroes are not a good idea, better put an extra space if you want to align it

Reply:
Just to clarify @ThiefMaster's comment:
```python
>>> 012
10```
(octal)

Reply:
And they are a syntax error in Python 3, due to how easy it was to mess things up by not realizing that the 0-prefix created an octal number.

Reply:
Thanks, I actually wanted to avoid leading zeros, I think I just overlooked that one.